bigquery_etl/util.py

```diff
@@ -142,13 +190,33 @@ def load_views(bq: bigquery.Client, dataset: bigquery.Dataset, views: Dict[str,
 
 def read(*paths: str, decoder: Optional[Callable] = None, **kwargs):
     """Read a file and apply decoder if provided."""
-    with open(os.path.join(*paths), **kwargs) as f:
-        return decoder(f) if decoder else f.read()
+    filepath = os.path.join(*paths)
+    with open(filepath, **kwargs) as f:
+        return decoder(f, filepath) if decoder else f.read()
 
 
-def ndjson_load(file_obj: Iterable[str]) -> List[Any]:
+def ndjson_load(file_obj: Iterable[str], filepath: str) -> List[Any]:
     """Decode newline delimited json from file_obj."""
-    return [json.loads(line) for line in file_obj]
+    res = []
+    for i, line in enumerate(file_obj):
+        try:
+            res.append(json.loads(line))
+        except json.JSONDecodeError as e:
+            raise NDJsonDecodeError(
+                f"Line {i+1} column {e.colno} of file {filepath}, {e.msg}"
+            )
+
+    return res
+
+
+def json_load(file_obj: Iterable[str], filepath: str) -> List[Any]:
+    """Decode json from file_obj."""
+    try:
+        return json.loads("".join(file_obj))
```

Comment:
this should use the json method for loading directly from file objects, and fix the parameter type of file_obj to be `io.TextIOWrapper`
```suggestion
        return json.load(file_obj)
```